#5216 update Markdown config doc to API changes

TimKam opened the pull request:
Subject: Update Markdown config doc to API changes

### Feature or Bugfix
- Docs
### Purpose
- Updates Markdown configuration documentation to reflect the changes to the source parser API.

### Detail
- Depends on update of *recommonmark*: https://github.com/rtfd/recommonmark/pull/113
- Once the *recommonmark* is released, I can specify it's exact minimal version that supports the changes and we can merge this PR.

### Relates
- #5216

TimKam committed:
#5216 update Markdown config doc to API changes

TimKam committed:
#5216 change recommonmark installation instruction

the pypi package has not been updated since January 2016;
it is better to install from source

TimKam commented:
@tk0miya Can we merge this? :-)

tk0miya commented on doc/usage/markdown.rst:
```diff
@@ -22,26 +22,32 @@ Configuration
 
-   You can replace ``.md`` with a filename extension of your choice.
+   .. versionchanged:: 1.8
+      Version 1.8 deprecates and version 2.0 removes the ``source_parsers``
```
`source_parsers` will be removed in Sphinx-3.0, not 2.0.

tk0miya commented on doc/usage/markdown.rst:
```diff
@@ -22,26 +22,32 @@ Configuration
+#. Install the Markdown parser *recommonmark* from its source on GitHub::
 
-      pip install recommonmark
+      pip install git+https://github.com/rtfd/recommonmark
```
Unfortunately, recommonmark has not been released for a long time. So this example uses HEAD version.
But many users use released version. So we have to mention about it.

Hopefully, it would be nice if recommonmark team will ship a new release...

tk0miya reviewed (changes requested):
I worried about using unreleased version of recommonmark in our document. So some spetial mention would be needed.

TimKam committed:
#5216 add note regarding recommonmark version

TimKam committed:
#5216 fix removal note

TimKam commented:
@tk0miya I fixed the removal info and added a note that explains the recommonmark version issue. Recommonmark has not been released for years and nobody there followed up on my comment some months ago. I think it's okay to update our documentation now and adjust it in case recommonmark will be released.

TimKam committed:
Merge branch 'master' into 5216-update-markdown-doc

tk0miya reviewed (approved):
+1

tk0miya commented:
Let see when it will be released.
Thanks,
